_posts/2022-01-07-The-Basics-of-Machine-Learning-without-math/2022-01-07-The-Basics-of-Machine-Learning-without-math.md

```diff
@@ -0,0 +1,66 @@
+
+
+
+
+You give a input into it and through the interaction of the different neurons the result is formed
+
+## What is a Neuron
+
+As you could see in the image above there are 3 types of neurons.
+The green ones are input Neurons. The just have connections to their right and no other connections. These are used to input data into the neuronal network. <br>
+The second type are the blue ones. How many collumns of them exist is not given and can be changed. They have connections to their left (input) and to their right (output)
+They take the inputs, process them and then generate an output. <br>
+The last type of neurons are the yellow one. They are at the end of the network. Their Output is the output of the whole network. <br>
+
+
+All 3 types of the network can be descriped through one model
+
+<img src="1*O7YSSqlOdQuNgMrH7J_3dg.png" width="600"><br>
+https://miro.medium.com/max/428/1*O7YSSqlOdQuNgMrH7J_3dg.png<br><br>
+
+A neuron has multiple inputs. The inputs are the outputs of all neurons in the previous collum.
+These inputs get weights. For example if a input is 0.6 and the weight is 0.1 we take 0.6 * 0.1 and then we have the "real" input. Every input has its own weight. All of the inputs multiplied with the weigth are added together. This sum is now put into a function. The function determines how much output is generated for the given sum. Often the function looks like this:
+
+<img src="1*oePAhrm74RNnNEolprmTaQ.png" width="600"><br>
+https://miro.medium.com/max/357/1*oePAhrm74RNnNEolprmTaQ.png<br><br>
+The output of this function (in this case RELU) is the output of the neuron and therefore the input for the next collum of neurons.<br>
+If we have a description of the arrangement of the neurons and a list of all weights we can reproduce the network over and over with the same outputs for the same inputs. And these two informations are the core of a neuronal network. How the Neurons are arrenged and what the weights are. 
+
+## What is Backpropagation
+The structure of a network impacts its perfomance but the best variable to shape it are the weights. If we get the right weights its should be perfect. Obviously getting good weights is hard. I know i said this is without math but we need a little bit. A neuronal network is like a function with millions of parameters(weights) which produces a deterministic result. We can imagen this like a 3D map. 
+
+
+<img src="gradient-descent-local-minima.png" width="600"><br>
+https://bdtechtalks.com/wp-content/uploads/2020/04/gradient-descent-local-minima.png <br><br>
+The lower we get the better our network performs. Finding the lowest point overall (Global Minima) is at the moment impossible. We take a other approach. We start at a random point somewhere. So we just use random weights. Luckily we have a way to find the way which leads us from our point. This is called negative gradiant. If you are intereseted in math you can research this on your own. <br>
```

Comment:
```suggestion
The lower we get the better our network performs. Finding the lowest point overall (Global Minima) is generally hard, as it is hard to express the desired function at all. We take a other approach. We start at a random point somewhere, so we just use random weights. Luckily we have a way to find the way which leads us from our point. This is called negative gradient. If you are interested in math you can research this on your own. <br>
```

Why do we need to get lower? You likely need to introduce some notion of "Error" for this, and that we can measure this error. And *then* we can say that we want to minimize the error.

Otherwise - why should I care about getting to the lowest point of a function?